src/proof/proof_rule.h

```diff
@@ -1385,241 +1385,349 @@ enum class PfRule : uint32_t
+   *
+   * .. math::
+   *
+   *   \inferrule{t\in R\mid -}{\texttt{RegExpOpr::reduceRegExpPos}(t\in R)}
+   *
+   * corresponding to the one-step unfolding of the premise.
+   * \endverbatim
+   */
   RE_UNFOLD_POS,
-  // ======== Regular expression unfold negative
-  // Children: (P:(not (str.in.re t R)))
-  // Arguments: none
-  // ---------------------
-  // Conclusion:(RegExpOpr::reduceRegExpNeg((not (str.in.re t R)))),
-  // corresponding to the one-step unfolding of the premise.
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Regular expressions -- Negative Unfold**
+   *
+   * .. math::
+   *
+   *   \inferrule{t\not\in R\mid -}{\texttt{RegExpOpr::reduceRegExpNeg}(t\not\in R)}
+   *
+   * corresponding to the one-step unfolding of the premise.
+   * \endverbatim
+   */
   RE_UNFOLD_NEG,
-  // ======== Regular expression unfold negative concat fixed
-  // Children: (P:(not (str.in.re t R)))
-  // Arguments: none
-  // ---------------------
-  // Conclusion:(RegExpOpr::reduceRegExpNegConcatFixed((not (str.in.re t
-  // R)),L,i)) where RegExpOpr::getRegExpConcatFixed((not (str.in.re t R)), i) =
-  // L. corresponding to the one-step unfolding of the premise, optimized for
-  // fixed length of component i of the regular expression concatenation R.
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Regular expressions -- Unfold negative concatenation, fixed**
+   *
+   * .. math::
+   *
+   *   \inferrule{t\not\in R\mid
+   *   -}{\texttt{RegExpOpr::reduceRegExpNegConcatFixed}(t\not\in R,L,i)}
+   *
+   * where :math:`\texttt{RegExpOpr::getRegExpConcatFixed}(t\not\in R, i) = L`,
+   * corresponding to the one-step unfolding of the premise, optimized for fixed
+   * length of component :math:`i` of the regular expression concatenation
+   * :math:`R`.
+   * \endverbatim
+   */
   RE_UNFOLD_NEG_CONCAT_FIXED,
-  // ======== Regular expression elimination
-  // Children: none
-  // Arguments: (F, b)
-  // ---------------------
-  // Conclusion: (= F strings::RegExpElimination::eliminate(F, b))
-  // where b is a Boolean indicating whether we are using aggressive
-  // eliminations. Notice this rule concludes (= F F) if no eliminations
-  // are performed for F.
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Regular expressions -- Elimination**
+   *
+   * .. math::
+   *
+   *   \inferrule{-\mid F,b}{F =
+   *   \texttt{strings::RegExpElimination::eliminate}(F, b)}
+   *
+   * where :math:`b` is a Boolean indicating whether we are using aggressive
+   * eliminations. Notice this rule concludes :math:`(= F F)` if no eliminations
+   * are performed for :math:`F`.
+   * \endverbatim
+   */
   RE_ELIM,
-  //======================== Code points
-  // Children: none
-  // Arguments: (t, s)
-  // ---------------------
-  // Conclusion:(or (= (str.code t) (- 1))
-  //                (not (= (str.code t) (str.code s)))
-  //                (not (= t s)))
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Code points**
```

Comment:
```suggestion
   * **String -- Code points**
```
Also in the other rules...

Reply:
Also: what does this rule mean? Are `t` and `s` different characters?

Reply:
`t` and `s` are arbitrary integer terms.  This rule says that the mapping from integer code points to length one strings (there are no characters in SMT-LIB) is injective.

Reply:
We should actually use `to_code` instead of `code` here, the original rule didn't use proper SMT-LIB syntax.

Reply:
> We should actually use `to_code` instead of `code` here, the original rule didn't use proper SMT-LIB syntax.

Done

Reply:
> Also in the other rules...

@nafur Do you mean for the regular expression and sequence ones? I was uncertain whether to add the strings prefix for them... what do you think @ajreynol ?

Reply:
Add `STRINGS_` to the proof rule names you mean? Or is this purely a documentation thing?

Reply:
Purely a documentation thing. It's an informal nampespace basically

Reply:
FYI I have considered being more pedantic about proof rule names so the CONCAT_X is STRING_CONCAT_X ...  the downside it is quite verbose.

Reply:
> Then yes, all should be under strings.

OK. 

> FYI I have considered being more pedantic about proof rule names so that CONCAT_X is STRING_CONCAT_X ... the downside it is quite verbose.

I'm not sure we want to go there... I think unless there is ambiguity it's fine to leave like this.